src/transformers/keras_callbacks.py

```diff
@@ -160,7 +183,7 @@ def on_epoch_end(self, epoch, logs=None):
                 labels = None
             if self.predict_with_generate:
                 if isinstance(batch, dict):
-                    generation_inputs = batch[self.main_input_name]
+                    generation_inputs = batch["input_ids"]
```

Comment:
This seems like a bad change.

Reply:
Ah, sorry, this was a change I made in testing! I'll put the `main_input_name` code back, you're right.

Reply:
Fixed!